changelogs/fragments/81762-no_log.yml

```
@@ -0,0 +1,2 @@
+bugfixes:
+  - argument spec validation - make sure that all ``no_log`` parameters are processed even if some arguments have values of the wrong type (https://github.com/ansible/ansible/pull/81672).
```

Since this is just a preventative measure, this should probably be a minor_changes fragment.
```suggestion
minor_changes:
  - argument spec validation - process all ``no_log`` parameters recursively before validating the argument spec (https://github.com/ansible/ansible/pull/81672).
```

---

Once #81631 or something equivalent is merged, this fixes a real CVE-worthy bug.

The only reason this isn't a bug right now is that a crash in another part is preventing this bug to have any effect. It is still a bug IMO.